### PR TITLE
Reply UI

### PR DESCRIPTION
**Pull request by andrewcchau**
Added reply button + UI for status responses

### CODE REVIEW

File: src/js/components/UserTimeline.js [hunk 1]
```diff
@@ -32,11 +36,22 @@ class UserTimeline extends React.Component {
         }
     }
 
+    displayModal(toggle, content){
+        let modalContent;
+        if(content) {
```
Comment (kev-chan) [inline src/js/components/UserTimeline.js]: you can optimize as this conditional doesn't do anything.  If the content is not passed in, the content variable is 'undefined', the same as modalContent

File: src/js/services/httpCall.js [hunk 2]
```diff
@@ -37,5 +37,21 @@ const PostToTwitter = (content, callback) => {
...
 
+const ReplyToTweet = (content, callback) => {
+    if(content) {
+        return fetch('http://localhost:8080/api/1.0/twitter/tweet/reply', {
+            method: 'POST',
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/x-www-form-urlencoded"
+            },
+            body: encodeURIComponent('statusID') + "=" + encodeURIComponent(content.statusID)
```
Comment (kev-chan) [inline src/js/services/httpCall.js]: this works.  Let's pass in json instead for any post bodies.  Please do the same for the PostTweet as well

File: src/js/components/UserTimeline.js (hunk 1 above)
Comment (kev-chan) [reply src/js/components/UserTimeline.js]: btw, what is content/modelContent represent?

File: src/js/components/Modal.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    componentDidMount() {
+        let tbClass = document.getElementsByClassName("modalTextArea");
+        this.setState({
+            textBox: tbClass && tbClass[0]
+        })
+    }
+
+    updateUI() {
+        if(this.state.textBox && this.state.textBox.value) {
```
Comment (kev-chan) [inline src/js/components/Modal.js]: can be simplified to the following i believe
this.setState({
modalMsgLength: this.state.textBox && this.state.textBox.value && this.state.textBox.length || 0

File: src/js/components/Modal.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            this.setState({
+                modalMsgLength: this.state.textBox.value.length
+            });
+        } else {
+            this.setState({
+                modalMsgLength: 0
+            })
+        }
+
+        this.setState({
```
Comment (kev-chan) [inline src/js/components/Modal.js]: any reason why setState is called twice?

File: src/js/components/Modal.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        this.state.textBox.value = '';
+        this.setState({
+            modalMsgLength: 0
+        })
+    }
+
+    updateReturnMessage(statusCode) {
+        if(statusCode) {
+            let success;
+            if(statusCode == 200) {
```
Comment (kev-chan) [inline src/js/components/Modal.js]: success = statusCode == 200;

File: src/js/components/PostTweetUI.js [hunk 6]
```diff
@@ -1,33 +1,33 @@
...
+        })
     }
 
-    updateReturnMessage(message) {
-        if(message) {
+    updateReturnMessage(statusCode) {
+        if(statusCode) {
             let success;
-            if(_.isEqual(message, 200)) {
+            if(statusCode == 200) {
```
Comment (kev-chan) [inline src/js/components/PostTweetUI.js]: same as previous statusCode comment

File: src/js/components/Modal.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                success = true;
+            } else {
+                success = false;
+            }
+
+            this.setState({
+                returnMessage: (success ? "Replied Successfully!" : "Failed to Reply!"),
+                retAppend: (success ? " success" : " error")
```
Comment (kev-chan) [inline src/js/components/Modal.js]: what is retAppend?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/js/components/UserTimeline.js (hunk 1 above)
Comment (andrewcchau) [reply src/js/components/UserTimeline.js]: Content/modalContent represents the tweet to be replied to. I've renamed it to "tweet" as it seems more fitting

File: src/js/components/Modal.js (hunk 7 above)
Comment (andrewcchau) [reply src/js/components/Modal.js]: A string to append to the append message component. Mainly there to change colors using css -> "___.success" { color: green } and "____.error" { color: red }. Will rename for better readability

**Pull request merged**